tests/snappi_tests/multidut/ecn/test_multidut_ecn_marking_with_snappi.py

```diff
@@ -0,0 +1,107 @@
+                                get_snappi_ports,     # noqa: F811
+                                tbinfo,      # noqa: F811
+                                multidut_port_info,     # noqa: F811
+                                prio_dscp_map):                    # noqa: F811
+    """
+    Verify ECN marking both pre and post port shut/no shut toggle
+
+    Args:
+        request (pytest fixture): pytest request object
+        snappi_api (pytest fixture): SNAPPI session
+        conn_graph_facts (pytest fixture): connection graph
+        fanout_graph_facts (pytest fixture): fanout graph
+        duthosts (pytest fixture): list of DUTs
+        lossless_prio_list (pytest fixture): list of all the lossless priorities
+        prio_dscp_map (pytest fixture): priority vs. DSCP map (key = priority).
+        prio_dscp_map (pytest fixture): priority vs. DSCP map (key = priority).
+        tbinfo (pytest fixture): fixture provides information about testbed
+        get_snappi_ports (pytest fixture): gets snappi ports and connected DUT port info and returns as a list
+    Returns:
+        N/A
+    """
+
+    for testbed_subtype, rdma_ports in multidut_port_info.items():
+        tx_port_count = 2
+        rx_port_count = 1
+        snappi_port_list = get_snappi_ports
+        pytest_assert(MULTIDUT_TESTBED == tbinfo['conf-name'],
+                      "The testbed name from testbed file doesn't match with MULTIDUT_TESTBED in variables.py ")
+        pytest_assert(len(snappi_port_list) >= tx_port_count + rx_port_count,
+                      "Need Minimum of {} ports defined in ansible/files/*links.csv file".
+                      format(tx_port_count + rx_port_count))
+
+        pytest_assert(len(rdma_ports['tx_ports']) >= tx_port_count,
+                      'MULTIDUT_PORT_INFO doesn\'t have the required Tx ports defined for \
+                      testbed {}, subtype {} in variables.py'.
+                      format(MULTIDUT_TESTBED, testbed_subtype))
+
+        pytest_assert(len(rdma_ports['rx_ports']) >= rx_port_count,
+                      'MULTIDUT_PORT_INFO doesn\'t have the required Rx ports defined for \
+                      testbed {}, subtype {} in variables.py'.
+                      format(MULTIDUT_TESTBED, testbed_subtype))
+
+        # Collect port names from rx_ports and tx_ports into a set for uniqueness
+        all_ports_set = set(port['port_name'] for port in rdma_ports['rx_ports'] + rdma_ports['tx_ports'])
+        pytest_assert(len(all_ports_set) >= tx_port_count + rx_port_count,
+                      'MULTIDUT_PORT_INFO doesn\'t have at least {} unique ports for \
+                      testbed {}, subtype {} in variables.py'.
+                      format(tx_port_count + rx_port_count, MULTIDUT_TESTBED, testbed_subtype))
+
+        logger.info('Running test for testbed subtype: {}'.format(testbed_subtype))
+
+        if is_snappi_multidut(duthosts):
```

Comment:
Raghav has a recent commit which moved these port selection into fixture for easy setup.  https://github.com/sonic-net/sonic-mgmt/pull/15099/

Do you want to adopt the change in this PR or address it in another PR?

Reply:
integrated the change